src/zelos/ext/platforms/linux/syscalls/syscalls.py

```diff
@@ -610,7 +610,10 @@ def sys_readlink(sm, p):
     # allowed without doing a validation in the filesystem first
     try:
         pathname = p.memory.read_string(args.pathname)
-        linked_path = os.readlink(pathname)
+        if pathname == "/proc/self/exe":
```

Comment:
This code requires 2 todos

TODO: support symbolic links in emulated filesystem.
TODO: sanitize request to readlink